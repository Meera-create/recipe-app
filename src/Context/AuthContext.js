import React, { createContext, useState, useEffect } from "react";
import { onAuthStateChanged } from "firebase/auth";
import { auth } from "../config/firebase";

export const Context = createContext();

export const AuthContext = ({ children }) => {
  const [user, setUser] = useState(null);
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    const unsubscribe = onAuthStateChanged(auth, (currentUser) => {
      setLoading(false);
      if (currentUser) {
        setUser(currentUser);
      } else {
        setUser(null);
      }
    });
    return () => {
      if (unsubscribe) unsubscribe();
    };
  }, []);

  const values= {user, setUser,};

  return (
    <Context.Provider value={values}>{!loading && children}</Context.Provider>
  )
}