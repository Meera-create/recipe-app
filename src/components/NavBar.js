import React, { useContext } from "react";
import { NavLink, Link } from "react-router-dom";
import Logout from "./Logout";
import { Context } from "../Context/AuthContext";
import '../styles/components/_nav-bar.scss'
import image from '../../src/styles/images/the pantry (1).jpg'

const NavBar = () => {
  const { user } = useContext(Context);

  return (
    <div className="navbar">
      {user && (
        <div className="buttons">
          <div className="boxes">
            <Link reloadDocument to="/recipe-form">
              <img alt="logo" src={image} />
            </Link>
          </div>
          <div className="boxes2">
            <button type="button" className="add-recipe">
              <NavLink to="add-recipe">Add Recipe</NavLink>
            </button>
            <button type="button"className="my-account">
              <NavLink to="my-account">My Account</NavLink>
            </button>
            <Logout />
          </div>
        </div>
      )}
    </div>


    // <div className="navbar">
    //   <Link reloadDocument to="/recipe-form">
    //     <img alt="logo" src={image} />
    //     {/* <FaPoop /> */}
    //   </Link>
    //   {user && (
    //     <ul className="navbar_buttons">
    //       <li className="account_button">
    //         <button type="button"className="my-account">
    //           <NavLink to="my-account">My Account</NavLink>
    //         </button>
    //       </li>
    //       <li className="logout_button">
    //         <Logout />
    //       </li>
    //     </ul>
    //   )}
    // </div>
  )
};

export default NavBar;