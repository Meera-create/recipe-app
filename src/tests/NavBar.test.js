import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import NavBar from "../components/NavBar";
import { AuthContext } from "../Context/AuthContext";  // Import the real AuthContext

describe("NavBar", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <AuthContext> {/* Use the real AuthContext here */}
          <NavBar />
        </AuthContext>
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
