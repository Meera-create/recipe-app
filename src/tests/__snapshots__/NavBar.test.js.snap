// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders correctly 1`] = `<DocumentFragment />`;
